<launch>

    <include file="$(find safe_traffic_weaving)/launch/multi_velodyne_points.launch" />

    <node pkg="nodelet" type="nodelet" name="pcl_manager" args="manager" output="screen" />

    <node pkg="nodelet" type="nodelet" name="M_intensity_filter" args="load pcl/PassThrough pcl_manager" output="screen">
        <remap from="~input" to="/M_HDL32/velodyne_points" />
        <rosparam>
            filter_field_name: intensity
            filter_limit_min: 50
            filter_limit_max: 1000
            filter_limit_negative: False
        </rosparam>
    </node>

    <node pkg="nodelet" type="nodelet" name="FL_intensity_filter" args="load pcl/PassThrough pcl_manager" output="screen">
        <remap from="~input" to="/FL_VLP16/velodyne_points" />
        <rosparam>
            filter_field_name: intensity
            filter_limit_min: 50
            filter_limit_max: 1000
            filter_limit_negative: False
        </rosparam>
    </node>

    <node pkg="nodelet" type="nodelet" name="FR_intensity_filter" args="load pcl/PassThrough pcl_manager" output="screen">
        <remap from="~input" to="/FR_VLP16/velodyne_points" />
        <rosparam>
            filter_field_name: intensity
            filter_limit_min: 50
            filter_limit_max: 1000
            filter_limit_negative: False
        </rosparam>
    </node>

    <node
        pkg="tf"
        type="static_transform_publisher"
        name="X1_to_vehicle_base"
        args="0 0 0 0 0 0 1 x1 vehicle_base 100"
    />

    <node
        pkg="safe_traffic_weaving"
        type="viz_vehicle.py"
        name="tracked_object_visualization"
        args="tracked_object 3.79 1.87 0.095 blue"
    />

    <node
        pkg="safe_traffic_weaving"
        type="multi_velodyne_object_tracking.py"
        name="object_tracking"
        args="100"
    />

</launch>
